### PR TITLE
fix(Stack): fix spacing compat for chrome under 84

### PR DESCRIPTION
**Pull request by MarvelSQ**
## ISSUES

`spacing` under **chrome 71**
截屏2022-09-14 上午9 58 22

flexbox-gap compat
- Chrome 84+
- Safari 14.1+
> compatibility table https://caniuse.com/flexbox-gap

## CHANGES
add `isSupportFlexGap` for Stack to check browser support

### CODE REVIEW

File: src/utils/BrowserDetection.ts [hunk 1]
```diff
@@ -14,3 +14,47 @@ export const isIE11 = () =>
 
 // Edge 20+
 export const isEdge = () => canUseDOM && !isIE() && 'styleMedia' in window;
+
+export const getChromeVersion = () => {
+  if (canUseDOM) {
+    const match = window.navigator.userAgent.match(/Chrome\/([0-9]+)\./);
+    return match ? parseFloat(match[1]) : false;
```
Comment (simonguo) [inline src/utils/BrowserDetection.ts:21]: Chromium is used in linux desktop systems.

```suggestion
    const match = window.navigator.userAgent.match(/Chrom(e|ium)\/([0-9]+)\./); 
    return match ? parseFloat(match[2]) : false;
```

File: src/utils/BrowserDetection.ts [hunk 2]
```diff
@@ -14,3 +14,47 @@ export const isIE11 = () =>
...
+
+  return false;
+};
+
+export const getSafariVersion = () => {
+  if (canUseDOM) {
+    const match =
+      !!window.navigator.userAgent.match(/Safari\//) &&
+      window.navigator.userAgent.match(/Version\/([0-9]+)\./);
+    return match ? parseFloat(match[1]) : false;
```
Comment (simonguo) [inline src/utils/BrowserDetection.ts:32]: image

I recommend using `window.navigator.userAgent.match(/Version\/([\d\.]+).*Safari/) `. The version information it gets is more accurate. But I'm not sure if it has compatibility issues.

**Review by simonguo — CHANGES_REQUESTED** [review]
Add tests as new code whenever possible。

File: src/utils/BrowserDetection.ts (hunk 2 above)
Comment (MarvelSQ) [reply src/utils/BrowserDetection.ts:30]: there is a similar regexp from [detect-browser](https://github.com/DamonOehlman/detect-browser), the difference is their version support `_`. I think the current regexp works fine.
> https://github.com/DamonOehlman/detect-browser/blob/master/src/index.ts#L166

Comment (simonguo) [reply src/utils/BrowserDetection.ts:30]: 👌

**Pull request merged**